src/map/storage.c

```diff
@@ -212,21 +231,29 @@ int storage_additem(struct map_session_data* sd, struct item* item_data, int amo
  *------------------------------------------*/
 int storage_delitem(struct map_session_data* sd, int n, int amount)
 {
+	struct item *it = NULL;
+
 	nullpo_retr(1, sd);
-	Assert_retr(1, n >= 0 && n < MAX_STORAGE);
-	if( sd->status.storage.items[n].nameid == 0 || sd->status.storage.items[n].amount < amount )
-		return 1;
 
-	sd->status.storage.items[n].amount -= amount;
-	if( sd->status.storage.items[n].amount == 0 )
-	{
-		memset(&sd->status.storage.items[n],0,sizeof(sd->status.storage.items[0]));
-		sd->status.storage.storage_amount--;
-		if( sd->state.storage_flag == STORAGE_FLAG_NORMAL )
-			clif->updatestorageamount(sd, sd->status.storage.storage_amount, MAX_STORAGE);
+	Assert_retr(1, n >= 0 && n < VECTOR_LENGTH(sd->storage.item));
+
+	nullpo_retr(1, (it = &VECTOR_INDEX(sd->storage.item, n)));
```

Comment:
no need to use `nullpo_retr()` here, `VECTOR_INDEX()` can't return `NULL` (well, it theoretically can, but only for `n == 0 && VECTOR_DATA(foo) == NULL`, but that can't be the case)